test/banned_deps_test.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+)
+
+var bannedImports = []string{
+	"github.com/knative",
```

Comment:
Why not using a simple `find` command in `post_build_tests()` in presubmit_tests.sh?

Reply:
because that is bash and no one uses that locally.

Reply:
Ok, given your latest commit it would have to be actually 3 `find`s. Or a loop. :)

Reply:
Developers should, because it would save their time when presubmits run on Prow. But you could also delegate the directory traversal to `find` in your go test. Anyway, I'm not gonna argue here, use whatever approach you think will be quicker adopted by the community.

Reply:
I never run the scripts locally because they do not tend to work on mac.